pkg/streamingpromql/testdata/ours/functions.test

```diff
@@ -67,22 +67,65 @@ eval range from 0 to 6m step 1m floor(some_metric)
+  some_metric{foo="bar"} 0 1 2 3 3 3 {{schema:3 sum:4 count:4 buckets:[1 2 1]}} {{schema:3 sum:5 count:3 buckets:[2 5 4]}}
 
 eval range from 0 to 7m step 1m last_over_time(some_metric[5s])
-  some_metric 0 1 2 3 _ _ {{schema:3 sum:4 count:4 buckets:[1 2 1]}} {{schema:3 sum:4 count:4 buckets:[1 2 1]}}
+  some_metric{foo="bar"} 0 1 2 3 _ _ {{schema:3 sum:4 count:4 buckets:[1 2 1]}} {{schema:3 sum:5 count:3 buckets:[2 5 4]}}
 
 eval range from 0 to 7m step 1m present_over_time(some_metric[3m])
-  {} 1 1 1 1 1 1 1 1
+  {foo="bar"} 1 1 1 1 1 1 1 1
 
 eval range from 0 to 7m step 1m present_over_time(some_metric[5s])
-  {} 1 1 1 1 _ _ 1 1
+  {foo="bar"} 1 1 1 1 _ _ 1 1
+
+eval range from 0 to 7m step 1m min_over_time(some_metric[3m])
+  {foo="bar"} 0 0 0 0 1 2 3 _
+
+eval range from 0 to 7m step 1m min_over_time(some_metric[5s])
+  {foo="bar"} 0 1 2 3 _ _ _ _
+
+eval range from 0 to 7m step 1m max_over_time(some_metric[3m])
+  {foo="bar"} 0 1 2 3 3 3 3 _
+
+eval range from 0 to 7m step 1m max_over_time(some_metric[5s])
+  {foo="bar"} 0 1 2 3 _ _ _ _
+
+eval_warn range from 0 to 10m step 1m sum_over_time(some_metric[3m])
+  {foo="bar"} 0 1 3 6 6 5 _ {{schema:3 sum:9 count:7 buckets:[3 7 5]}} {{schema:3 sum:9 count:7 buckets:[3 7 5]}} {{schema:3 sum:9 count:7 buckets:[3 7 5]}} {{schema:3 sum:5 count:3 buckets:[2 5 4]}}
+
+eval range from 0 to 5m step 1m sum_over_time(some_metric[3m])
+  {foo="bar"} 0 1 3 6 6 5
+
+eval range from 7m to 10m step 1m sum_over_time(some_metric[3m])
+  {foo="bar"} {{schema:3 sum:9 count:7 buckets:[3 7 5]}} {{schema:3 sum:9 count:7 buckets:[3 7 5]}} {{schema:3 sum:9 count:7 buckets:[3 7 5]}} {{schema:3 sum:5 count:3 buckets:[2 5 4]}}
+
+eval range from 0 to 7m step 1m sum_over_time(some_metric[5s])
+  {foo="bar"} 0 1 2 3 _ _ {{schema:3 sum:4 count:4 buckets:[1 2 1]}} {{schema:3 sum:5 count:3 buckets:[2 5 4]}}
+
+eval range from 0 to 2m step 1m sum_over_time(some_nhcb_metric[3m])
+  {baz="bar"} {{schema:-53 sum:1 count:5 custom_values:[5 10] buckets:[1 4]}} {{schema:-53 sum:16 count:7 custom_values:[5 10] buckets:[1 6]}} {{schema:-53 sum:19 count:22 custom_values:[5 10] buckets:[8 14]}}
+
+eval_warn range from 0 to 10m step 1m avg_over_time(some_metric[3m])
+  {foo="bar"} 0 0.5 1 1.5 2 2.5 _ {{schema:3 sum:4.5 count:3.5 buckets:[1.5 3.5 2.5]}} {{schema:3 sum:4.5 count:3.5 buckets:[1.5 3.5 2.5]}} {{schema:3 sum:4.5 count:3.5 buckets:[1.5 3.5 2.5]}} {{schema:3 sum:5 count:3 buckets:[2 5 4]}}
+
+eval range from 0 to 5m step 1m avg_over_time(some_metric[3m])
+  {foo="bar"} 0 0.5 1 1.5 2 2.5
+
+eval range from 7m to 10m step 1m avg_over_time(some_metric[3m])
+  {foo="bar"} {{schema:3 sum:4.5 count:3.5 buckets:[1.5 3.5 2.5]}} {{schema:3 sum:4.5 count:3.5 buckets:[1.5 3.5 2.5]}} {{schema:3 sum:4.5 count:3.5 buckets:[1.5 3.5 2.5]}} {{schema:3 sum:5 count:3 buckets:[2 5 4]}}
+
+eval range from 0 to 7m step 1m avg_over_time(some_metric[5s])
```

Comment:
We should have a couple of tests for `avg_over_time` that includes infinity.

We should also specifically test for a series that has infinity followed by NaN. Otherwise there is no point in continuing to check the points in the avg function as opposed to just returning infinity once we encounter it.

Reply:
> We should have a couple of tests for `avg_over_time` that includes infinity.

The tests from https://github.com/grafana/mimir/pull/8934/files#diff-e7d68a19a178199a4b1b50a87a3d5603044c971e736e361e1eb68738c67226d2R775 onwards include test cases with infinity - does this cover what you had in mind?

> We should also specifically test for a series that has infinity followed by NaN. Otherwise there is no point in continuing to check the points in the avg function as opposed to just returning infinity once we encounter it.

Good point, added in 4de9410.

Reply:
Yep, that covers it, and lgtm, thanks.